src/server/editorServices.ts

```diff
@@ -1192,7 +1192,7 @@ namespace ts.server {
             let hasChanges = this.pendingEnsureProjectForOpenFiles;
             this.pendingProjectUpdates.clear();
             const updateGraph = (project: Project) => {
-                hasChanges = updateProjectIfDirty(project) || hasChanges;
+                hasChanges = !updateProjectIfDirty(project) || hasChanges;

```

Comment:
This is incorrect change.. the intension here is to call updateGraph and get its return value only if it is marked as dirty.. So original code is correct in what it intends to do..

Reply:
Ok thanks for confirming. I'll abandon this PR. It anyway did not show any change in correctness. Though it's a little unfortunate that this change does not trigger any unit test failures.